docs/fasttext-notes.md

```diff
@@ -0,0 +1,152 @@
+FastText Notes
```

Comment:
Formally, there's no final goal. This is mostly for improving my personal understanding of our FastText implementation. It may be helpful to others who follow in my footsteps later.

We can use this to create higher-level design documentation when the refactoring is finished. I don't think it's worth linking classes/methods yet, because those may change during the refactoring.

Reply:
Okay, in that case, let's stay it as is, but convert to `.rst` & include in docs when you are fully done with gensim